### PR TITLE
Argo data submission wf@0.3.0

### PR DESCRIPTION
**Pull request by edsu7**
Also feel free to drop params from recommended tests, pipeline should catch the errors.

Testing local file submission (register a new sample to test without `--skip_duplicate_check` param)
```
 nextflow run ../main.nf -params-file local-test-job-InputBam.json --api_token ${token} --skip_duplicate_check true
```

Multiple test scenario
- Metadata JSON payload to TSVs
- cram2bam
- Incorrect metadata (corrected by Sanity check)
```
nextflow run ../main.nf -params-file local-test-job-InputCramJson.json --api_token ${token}  --skip_duplicate_check true --ref_genome_fa hs37d5.fa.gz
```

Supporting ref genome
```
curl ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz -o hs37d5.fa.gz
samtools faidx hs37d5.fa.gz
```

### CODE REVIEW

File: argo-data-submission-wf/main.nf [hunk 1]
```diff
@@ -166,38 +189,103 @@ process checkCramReference{
...
         params.download_mode,
         file_info_tsv,
         checkCramReference.out.check_status
       )
       sequence_files=egaWf.out.sequence_files
+    } else if (og_file_info_tsv.startsWith("NO_FILE")){
+      sequence_files=file_info_tsv.splitCsv( header : true , sep:'\t').map( row -> file("${row.path}",checkIfExists : true)) 
     } else {
-      sequence_files=Channel.fromPath(file_info_tsv) | splitCsv( header : true , sep:'\t') | map( row -> file("${row.path}",checkIfExists : true)) 
+      sequence_files=Channel.fromPath(file_info_tsv).splitCsv( header : true , sep:'\t').map( row -> file("${row.path}",checkIfExists : true))
```
Comment (lindaxiang) [inline argo-data-submission-wf/main.nf:288]: Just curious, what's the reason for differential `og_file_info_tsv.startsWith("NO_FILE")`?

**Review by lindaxiang — APPROVED** [review]
Test all the jobs other than ega related.

File: argo-data-submission-wf/main.nf (hunk 1 above)
Comment (edsu7) [reply argo-data-submission-wf/main.nf:288]: Naming scheme was to track original `experiment.tsv` and updated version. The usage in `file_info` and `read_groups` was to stay consistent.

**Review by lindaxiang — APPROVED** [review]
Test and work as expect.

**Comment by edsu7** [comment]
LGTM

**Pull request merged**